File: 1636-sort-array-by-increasing-frequency/1636-sort-array-by-increasing-frequency.cs
public class Solution
{
  private static Dictionary<T, int> FindFrequency<T>(IList<T> list) where T : notnull
  {
    var result = new Dictionary<T, int>();
    foreach (var item in list)
    {
      result[item] = result.GetValueOrDefault(item, 0) + 1;
    }
    return result;
  }

  public int[] FrequencySort(int[] nums)
  {
    var freqs = FindFrequency(nums).ToArray();
    var sorted = freqs.OrderBy(a => a.Value).ThenByDescending(a => a.Key);
    var result = new int[nums.Length];
    int j = 0;
    for (int i = 0; i < sorted.Count(); i++)
    {
      var pair = sorted.ElementAt(i);
      for (int k = 0; k < pair.Value; k++)
      {
        result[j] = pair.Key;
        j++;
      }
    }
    return result;
  }
}

File: 1636-sort-array-by-increasing-frequency/1636-sort-array-by-increasing-frequency.rs
use std::collections::HashMap;

impl Solution {
    pub fn frequency_sort(nums: Vec<i32>) -> Vec<i32> {
        let mut freqs = HashMap::<i32, u8>::new();

        for num in nums.iter() {
            match freqs.get(num) {
                Some(freq) => {
                    freqs.insert(*num, *freq + 1);
                }
                None => {
                    freqs.insert(*num, 1);
                }
            }
        }

        let mut pair = Vec::<(&i32, &u8)>::new();

        for ele in freqs.iter() {
            pair.push(ele);
        }

        pair.sort_unstable_by(|a, b| {
            if a.1 != b.1 {
                a.1.cmp(b.1)
            } else {
                b.0.cmp(a.0)
            }
        });

        let mut result = Vec::<i32>::new();

        for (k, v) in pair {
            for _ in 0..*v {
                result.push(*k);
            }
        }

        result
    }
}
